BUILD.md

```diff
@@ -469,6 +407,19 @@ meson compile -C build
 ninja -C build
 ```
 
+### Other build types
```

Comment:
Meson build snippets appears stuck under the Auto FDO level.

Suggest a structure like:

 **./scripts/build.sh stuff ... ** 
   - Windows
   - MacOS
   - Linux
   - AutoFDO
 
 **meson stuff ... ** 
  - Standard build for everyone
  - Platform nuances
  - Specialist build typeS: sanitizers, LTO, test-cases, etc..

Also, regenerate or repopulate the table of contents after.

Or even split the two files:
 - BUILD-script.md
 - BUILD-meson.md

(then we can delete BUILD-script.md when Meson is running the show!)

Reply:
> (then we can delete BUILD-script.md when Meson is running the show!)

@kcgen but `build.sh` script does not work any more… Meson is already running the show on `meson` branch…

Since https://github.com/dosbox-staging/dosbox-staging/pull/890#issue-573009261:

> Con: **After this change, Autoconf buildsystem is no longer usable (!!!)**

Reply:
Ahh - in that case, knock it all out :-) and just go with 100% meson instructions in `BUILD.sh`

Reply:
(Small edit to comment above)

Reply:
The new approach of per-OS build notes is -perfect- @dreamer .

With the previous notes (and a bit in the README.md), I found myself scrolling through sections, having to skip stuff and memorize (or grep) specific lines.   For example, DOSBox on the Pi uses the console framebuffer, (RetroPie users don't have a Window manager); so now I can simply `cat` the file and type the commands verbatim.  Doesn't get easier than that!

For community members new to compiling, the short number of steps encourage one to try, versus the potential discouragement when one sees a many-page document.